core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java

```diff
@@ -55,14 +56,20 @@ public class TestExponentialGrowthPartitionMemoryEstimator
         throw new RuntimeException("should not be used");
     };
 
-    @Test
-    public void testDefaultInitialEstimation()
+    private ExponentialGrowthPartitionMemoryEstimator.Factory estimatorFactory;
```

Comment:
This is an anti-pattern. It makes the test class stateful and requires all tests to run in sequence. If you want to reuse the code to create the object, extract it into a method that creates a new instance it and returns it. Then call it from each of the individual test methods:

```java
private static ExponentialGrowthPartitionMemoryEstimator.Factory createEstimator()
{
    return new ExponentialGrowthPartitionMemoryEstimator.Factory(...);
}

@Test
public void testDefaultInitialEstimation()
{
    ExponentialGrowthPartitionMemoryEstimator.Factory estimatorFactory = createEstimator();
    ...
}
```

Reply:
Can you help me understand what's different here from every other `@BeforeEach` in the codebase?

Reply:
It's not, but that's a pattern we want to move away from wherever possible, and certainly avoid introducing new instances unless absolutely necessary.

Reply:
Rewrote as suggested.